src/fheroes2/ai/ai_hero_action.cpp

```diff
@@ -1651,6 +1655,18 @@ namespace AI
             const std::vector<Game::DelayType> delayTypes = { Game::CURRENT_AI_DELAY };
 
             while ( LocalEvent::Get().HandleEvents( !hideAIMovements && Game::isDelayNeeded( delayTypes ) ) ) {
+#if defined( WITH_DEBUG )
+                if ( HotKeyPressEvent( Game::HotKeyEvent::TRANSFER_CONTROL_TO_AI ) && Players::Get( hero.GetColor() )->isAIAutoControlMode() ) {
+                    if ( fheroes2::showMessage( fheroes2::Text( _( "Warning" ), fheroes2::FontType::normalYellow() ),
+                                                fheroes2::Text( _( "Do you want to regain control from AI?" ), fheroes2::FontType::normalWhite() ),
```

Comment:
I believe the text below was intended to be in this dialog since it's the regaining of the control which happens on the next turn.
`The effect will take place only on the next turn.`
```suggestion
                                                fheroes2::Text( _( "Do you want to regain control from AI? The effect will take place only on the next turn." ), fheroes2::FontType::normalWhite() ),
```